tests/tests_deployment/test_conda_store_scaling.py

```diff
@@ -0,0 +1,234 @@
+    log.info(f"Deployment logs: {messages}")
+    pod_names = find_conda_store_worker_pod_names()
+    if deployment.status.readyReplicas:
+        pod_name_lookup = messages.split('"')[1]
+        for n in pod_names:
+            if pod_name_lookup in n:
+                pod_name = n
+        api_response = client.CoreV1Api().read_namespaced_pod_log(
+            name=pod_name, namespace=NAMESPACE, container="conda-store-worker"
+        )
+        log.info(f"conda-store-worker logs: {api_response}")
+    return replica_count
+
+
+def find_conda_store_worker_pod_names():
+    """
+    find namespace pod msg
+    """
+    k8s_api_obj = client.CoreV1Api()
+    api_response = k8s_api_obj.list_namespaced_pod(NAMESPACE)
+    names = [
+        i.metadata.name
+        for i in api_response.items
+        if i.metadata.labels.get("role")
+        and "nebari-conda-store-worker" in i.metadata.labels["role"]
+    ]
+    return names
+
+
+@pytest.mark.timeout(20 * 60)
+def timed_wait_for_deployments(target_deployment_count, api_client):
+    log.info(
+        f"Waiting for deployments to reach target value {target_deployment_count}  ..."
+    )
+    replica_count = get_deployment_count(api_client)
+    while replica_count != target_deployment_count:
+        replica_count = get_deployment_count(api_client)
+        direction = "up" if target_deployment_count > replica_count else "down"
+        log.info(
+            f"Scaling {direction} deployments: from {replica_count} to {target_deployment_count}"
+        )
+        time.sleep(5)
+    log.info(f"Deployment count: {replica_count}")
+
+
+def delete_conda_environments(session):
+    existing_envs_url = f"https://{NEBARI_HOSTNAME}/{CONDA_STORE_API_ENDPOINT}/environment/?namespace=global"
+    response = session.get(existing_envs_url, verify=False)
+    for env in response.json()["data"]:
+        env_name = env["name"]
+        delete_url = f"https://{NEBARI_HOSTNAME}/{CONDA_STORE_API_ENDPOINT}/environment/global/{env_name}"
+        log.info(f"Deleting {delete_url}")
+        session.delete(delete_url, verify=False)
+    log.info("All conda environments deleted.")
+
+
+def create_conda_store_env(session):
+    _url = f"https://{NEBARI_HOSTNAME}/{CONDA_STORE_API_ENDPOINT}/specification/"
+    name = str(uuid.uuid4())
+    request_json = {
+        "namespace": "global",
+        "specification": f"dependencies:\n  - tqdm\nvariables: {{}}\nchannels: "
+        f"[]\n\ndescription: ''\nname: {name}\nprefix: null",
+    }
+    response = session.post(_url, json=request_json, verify=False)
+    log.debug(request_json)
+    log.debug(response.json())
+    return response.json()["data"]["build_id"]
+
+
+@pytest.mark.timeout(10)
+def build_n_environments(n, builds, session):
+    log.info(f"Building {n} conda environments.")
+    for _ in range(n):
+        time.sleep(1)
+        builds.append(create_conda_store_env(session))
+    return builds
+
+
+# TODO : remove filters
```

Comment:
Any info on when these filters can be removed?  Consider adding a comment or link explaining why they are skipped for now.